bredis/AsyncConnection.hpp

```diff
@@ -25,9 +25,11 @@
 
 namespace bredis {
 
+// Rename `S` to `NextLayer` or `AsyncStream` 
 template <typename S> class AsyncConnection {
     using protocol_type_t = typename S::protocol_type;
 
+    // What is the reason for constraining the type of S this way?
     static_assert(std::is_same<protocol_type_t, boost::asio::ip::tcp>::value ||
                       std::is_same<protocol_type_t,
                                    boost::asio::local::stream_protocol>::value,
```

Comment:
Redis uses either TCP or Unix Domain Sockets as transport layer. So the restriction is to prevent using UDP-sockets for here.